### ISSUE TITLE
Unable to sync tasks that depend on other tasks

### ISSUE THREAD

**Issue by iosengineer** [bradyt/taskw-dart#7]
<img src="https://user-images.githubusercontent.com/2148403/131784911-fe068b65-d2ec-4952-a6df-c4913b2a073d.png" width=49% height=49%><img src="https://user-images.githubusercontent.com/2148403/131784904-ce34d9c1-24d9-4b67-84fd-ac1fa2d2045b.png" width=49% height=49%>

**Comment by bradyt** [comment]
Thank you for bringing this to my attention.

It looks like a client sent an array to your Taskserver, which I failed to consider. Turns out, it is mentioned in the spec.

> Attribute: depends
>
> The "depends" field is a string containing a comma-separated unique set of UUIDs. If task 2 depends on task 1, then it is task 1 that must be completed first. Task 1 is considered a "blocking" tasks, and task 2 is considered a "blocked" task. For example:
>
>     "depends":",, ..."
>
> Note that in a future version of this specification, this will be changed to a JSON array of strings, like the "tags" field. 
>
> -- https://taskwarrior.org/docs/design/task.html#attr_depends

Is this from Taskwarrior 2.6.0 built from latest git? This might be from https://github.com/GothenburgBitFactory/taskwarrior/commit/20af583e21666d4825bfb81fcd1264c786bf4d01, on August 17th.

Or it might be from `rc.json.depends.array`, which apparently goes back to 2015.

I should be able to fix this pretty quickly. My plan is to modify `Task.fromJson` to be able to parse either of `"depends":"x,y,z"` or `"depends":["x","y","z"]`. And for now, it should be fine for `Task.toJson` to send as the first, `"depends":"x,y,z"`.

I think I just need to add a unit test to get it right, then I'll try to deploy a new version to App Store soon after.

**Comment by bradyt** [comment]
Potential fix for this issue will be in next release, `0.1.0`, which is in review stage at App Store, and I hope will be available in a few hours. I'll try to ping the thread again when it's available.

Aside: bump in minor version is due to https://github.com/bradyt/taskw-dart/issues/8.

The changes planned in previous comment are implemented at commit https://github.com/bradyt/taskw-dart/commit/b98921e9b8113b9debd1bc0846d1a7e994746fe2.

I expect that if you try sync'ing after updating to `0.1.0`, the sync will succeed virtually as if there were never a problem. The `backlog.data` will be sent with the same sync key and updates, including any new updates since the errors started, and Taskserver will recalculate the appropriate updates and sync key to send in response. If the app can now parse those updates, it will apply them to `$profile/.task/all.data`, and reset the `$profile/.task/backlog.data` with the new sync key.

**Comment by bradyt** [comment]
Version `0.1.0` is on App Store.

**Comment by bradyt** [comment]
I assume this is fixed. And fix will be on Play Store and F-Droid soon, as I'm in process of releasing `v0.1.1` to those and App Store.

Thanks again for letting me know about this. Please let me know if I missed something on this.

**Comment by iosengineer** [comment]
Thank you. Ah. I just realised why I didn't get notified about this. I got rid of the email account associated with my GitHub account and still haven't fixed that up. Ugh. The taskserver I'm running is compiled from the latest release git, yes. I just submitted another issue so I'll try to fix up my email shortly.
